File: app/Logo.js
/**
 * The kPlay mark.
 *
 * A lowercase "k" whose arm and leg are a single play triangle: the stem is
 * the letter, the triangle is both the rest of the letter and the play glyph.
 * One shape says the name and says what the app does, which is what the old
 * plain "K" tile could not.
 *
 * Drawn on a 48-unit grid so the same geometry can be restated as an Android
 * vector drawable (android-tv/app/src/main/res/drawable/ic_launcher_foreground.xml)
 * without redrawing it by eye.
 *
 * `size` is the box edge in pixels; the badge and the mark scale together.
 */
export default function Logo({ size = 32, className = '', rounded = 'rounded-lg' }) {
  return (
    <span
      className={`flex shrink-0 items-center justify-center bg-[var(--accent)] ${rounded} ${className}`}
      style={{ width: size, height: size }}
      aria-hidden="true"
    >
      <LogoMark size={size * 0.66} />
    </span>
  );
}

/** The bare glyph, no badge — for places that supply their own backdrop. */
export function LogoMark({ size = 24, color = '#fff' }) {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 48 48"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      aria-hidden="true"
    >
      {/* stem of the k */}
      <rect x="10" y="9" width="6" height="30" rx="3" fill={color} />
      {/* arm, leg and play triangle, all one shape */}
      <path d="M21 12.6 L39.2 24 L21 35.4 Z" fill={color} strokeLinejoin="round" />
    </svg>
  );
}

File: android-tv/app/src/main/java/com/kdrive/tv/ui/components/Logo.kt
package com.kdrive.tv.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.CornerRadius
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.kdrive.tv.ui.theme.K

/**
 * The kPlay mark.
 *
 * A lowercase "k" whose arm and leg are a single play triangle — the stem is
 * the letter, the triangle is both the rest of the letter and the play glyph.
 * It replaces the plain "K" tile, which said the name and nothing else.
 *
 * The geometry is the same 48-unit drawing used by the web mark (app/Logo.js)
 * and the launcher icon (res/drawable/ic_launcher_foreground.xml), scaled to
 * whatever size it is given, so the three never drift apart.
 */
@Composable
fun LogoMark(
    size: Dp,
    modifier: Modifier = Modifier,
    color: Color = K.TextPrimary,
) {
    Canvas(modifier.size(size)) {
        val unit = this.size.minDimension / 48f

        drawRoundRect(
            color = color,
            topLeft = Offset(10f * unit, 9f * unit),
            size = Size(6f * unit, 30f * unit),
            cornerRadius = CornerRadius(3f * unit, 3f * unit),
        )

        drawPath(
            path = Path().apply {
                moveTo(21f * unit, 12.6f * unit)
                lineTo(39.2f * unit, 24f * unit)
                lineTo(21f * unit, 35.4f * unit)
                close()
            },
            color = color,
        )
    }
}

/** The mark on its accent tile — the app's badge wherever one is wanted. */
@Composable
fun LogoBadge(size: Dp, modifier: Modifier = Modifier) {
    Box(
        modifier
            .size(size)
            .clip(RoundedCornerShape(size / 4.4f))
            .background(K.Accent),
        contentAlignment = Alignment.Center,
    ) {
        LogoMark(size = size * 0.66f)
    }
}
